ptftplib/dhcpserver.py

```diff
@@ -172,7 +172,13 @@ def __init__(self, pkt):
         # Strip off the ethernet frame and check the IP packet type. It should
         # be UDP (0x11)
         pkt = pkt[14:]
-        if ord(pkt[9]) != IP_UDP_PROTO:
+        # add adaptor for python3 and python2 
+        # for python2 the type of pkt[N] is str, while for python3 it is int
```

Comment:
Shouldn't this be resolved at a higher layer to ensure that we always pass in bytes and not strings to `DhcpPacket` ?

Reply:
In python3, the default format of network packet or file content is byte array, so the type of the element is int
In python2, it is str and str
This is the difference of design ideas for these 2 python language, not the principle of a single module.

Reply:
i have another better version, pls close this.